lib/paperclip/deprecations.rb

```diff
@@ -0,0 +1,34 @@
+require 'active_support/deprecation'
+
+module Paperclip
+  class Deprecations
+    class << self
+      def check
+        warn_aws_sdk_v1 if aws_sdk_v1?
+        warn_outdated_rails if active_record_version < '4.2'
+      end
+
+      private
+
+      def active_record_version
+        ::ActiveRecord::VERSION::STRING
+      end
+
+      def aws_sdk_v1?
+        defined?(::Aws) && aws_sdk_version < '2'
+      end
+
+      def warn_aws_sdk_v1
+        ActiveSupport::Deprecation.warn '[paperclip] [deprecation] AWS SDK v1 has been deprecated in paperclip 5. Please consider upgrading to AWS 2 before upgrading paperclip.'
```

Comment:
Line is too long. [177/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.